_posts/2023-06-05-timelines-save-time-and-money.md

```diff
@@ -0,0 +1,294 @@
+---
+layout: post
+title: "It’s Time for a Time-centric Stack: Timeline Abstractions Save Time and Money"
+image: assets/images/blog/Blog_Post_Time_Centric.png
+author: Brian Godsey
+---
+
+
+
+When processing and building real-time feature values from event-based data, it
+is easy to underestimate the power of using tools that are built to understand
+how time works. Many software tools can parse and manipulate time-based data,
+but the vast majority have little more than convenience functionality for
+parsing and doing arithmetic on date-time values. On the other hand, Kaskada has
+a built-in, natural understanding of time that just works by default, which can
+reduce development time and effort, simplify deployment and debugging, and
+reduce the compute and storage resources needed to support real-time event-based
+systems.
+
+In this post, I'll discuss some costly pitfalls when building data pipelines
+with event-based data. In particular, I will share some anecdotes from my past
+experience building SQL-based pipelines, how my team spent considerable time and
+money that (in hindsight) could have been avoided, and how Kaskada specifically
+could have simplified development and reduced costs. I will get into specifics
+below, but the lessons I've learned from using SQL on event data boil down to
+this: 
+
+> Event-based data and time values have special properties and natural
+> conventions, which lead us to write the same generic logic over and over. This
+> time-centric logic—using window functions, time-JOINs, etc—can become
+> complicated very quickly, both in code readability and computational
+> complexity. Because of this, writing time-centric logic is often inefficient
+> and buggy.
+
+The cautionary anecdotes that follow below may qualify as horror stories in some
+circles, but I promise that no humans or computing resources were harmed during
+the construction of these pipelines or in the mayhem that followed. There was,
+however, plenty of time and money lost on a few fronts.
+
+
+# Case study: calculating daily user statistics and aggregations
+
+In this example, I’d like to show how hard it can be to calculate (seemingly)
```

Comment:
Feels like it would be better to just jump into the example, and/or state what you show (rather than "like").